### PR TITLE
Adding Scala-Italy Discord server to community

### PR DESCRIPTION
**Pull request by TonioGela**
As @SethTisue asked me [here](https://discord.com/channels/632277896739946517/632277897448652844/904772814064734318), I've added the link to the Scala Italy Discord server and removed the deprecated gitter one.

### CODE REVIEW

File: community/index.md [hunk 1]
```diff
@@ -101,6 +100,7 @@ Other Scala-related Discord servers include:
 
 * **[Scalameta](https://discord.gg/RFpSVth)**: Scalameta-based tooling: Metals, Scalameta, Scalafix, Scalafmt, and Mdoc
 * **[Typelevel](https://discord.gg/XF3CXcMzqD)**: about the Typelevel ecosystem for pure-functional programming in Scala
+* **[ScalaItaly](https://discord.gg/8wadTgcZVt)**: the Italian Scala community Discord server
```
Comment (bishabosha) [inline community/index.md:103]: I'm not sure the position of the link should change (you can mention that it is Discord, like the `scala_ru` link mentions it is Telegram)

Comment (SethTisue) [reply community/index.md:103]: Agree

Comment (TonioGela) [reply community/index.md:103]: Okay, I will, move it there, thanks

Comment (TonioGela) [reply community/index.md]: I fixed it with the last commit. (It's worth squashing IMHO, I had to fix a typo)

**Pull request merged**